src/components/ReportsPage/HeatmapPage/HeatmapFiltersSelect.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 Cortex Applications, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Box, Button, FormControl, InputLabel, Select } from '@material-ui/core';
+import { Clear } from '@material-ui/icons';
+import React from 'react';
+
+interface ModalSelectProps<T> {
+  name: string,
+  value: T[],
+  onChange: (value: T[]) => void,
+  onReset: () => void,
+  options: React.ReactNode
+}
+
+export const ModalSelect = <T,>({ name, value, onChange, onReset, options }: ModalSelectProps<T>) => {
+  return (
+    <Box display="flex" flexDirection="row">
+      <FormControl variant="standard" fullWidth>
+        <InputLabel>{name}</InputLabel>
+        <Select
```

Comment:
The modal doesn't close when selecting an item, and it's also in front of the select itself. This is confusing and probably unexpected

https://github.com/cortexapps/backstage-plugin/assets/3069614/8abf52d0-3815-490e-922a-67222a904b1e

Reply:
This label being blocked is a problem for all usages of the select. I think resolving this in new task would be best.
Not closing on click is correct due to the field being multi-select.

Reply:
New task makes sense to me 👍